tests/testsuite/read_manifest.rs

```diff
@@ -0,0 +1,466 @@
+use cargo_test_macro::cargo_test;
+use cargo_test_support::registry::Package;
+use cargo_test_support::{main_file, project, str};
+
+use crate::ProjectExt;
+
+#[cargo_test]
+fn simple_with_deps() {
+    Package::new("a", "1.0.0")
+        .file("src/lib.rs", r#"pub fn f() -> i32 { 12 }"#)
+        .publish();
+
+    let p = project()
+        .file("src/main.rs", &main_file("Hello", &[]))
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "read-manifest-test"
+                version = "0.1.0"
+                authors = []
+                edition = "2024"
+
+                [dependencies]
+                a = "1.0.0"
+            "#,
+        )
+        .build();
+
+    p.cargo_plumbing("plumbing read-manifest")
+        .arg("--manifest-path")
+        .arg(p.root().join("Cargo.toml"))
+        .with_stderr_data(
+            str![[r#"
+[ERROR] unrecognized subcommand 'read-manifest'
+
+  tip: a similar subcommand exists: 'locate-manifest'
+
+Usage: cargo plumbing <COMMAND>
+
+For more information, try '--help'.
+
+"#]]
+        )
+        .with_status(2)
+        .run();
+
+    p.cargo_plumbing("plumbing read-manifest")
+        .arg("--manifest-path")
+        .arg(p.root().join("Cargo.toml"))
+        .arg("--workspace")
```

Comment:
Like you had before, `--workspace` should be added in the last commit so we can see how the behavior changes with it.  Seeing error to success is less interesting.

Reply:
Yes